dspace-oai/src/main/java/org/dspace/xoai/app/CCElementAdditional.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.xoai.app;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Item;
+import org.dspace.core.Context;
+import org.dspace.license.CCLookup;
+import org.dspace.license.factory.LicenseServiceFactory;
+import org.dspace.license.service.CreativeCommonsService;
+import org.dspace.xoai.util.ItemUtils;
+
+import com.lyncode.xoai.dataprovider.xml.xoai.Element;
+import com.lyncode.xoai.dataprovider.xml.xoai.Metadata;
+
+public class CCElementAdditional implements XOAIItemCompilePlugin {
```

Comment:
my understanding is that we need this enhancement also in DSpace 7, this add support for the CC license as required here
https://openaire-guidelines-for-literature-repository-managers.readthedocs.io/en/v4.0.0/field_licensecondition.html#

Reply:
That's fine, this is a new feature though, so it would need to be clarified as part of this PR and forward-ported to `master`.  My initial impression was the goal of this PR was just a backport from `master`, but as you note there are several new features embedded in this backport....each of those new features needs to either be `@deprecated` or forward ported back to `master`